src/goto-instrument/full_slicer.cpp

```diff
@@ -382,6 +382,13 @@ void full_slicer(
   full_slicert()(goto_functions, ns, a);
 }
 
+void full_slicer(goto_modelt &goto_model)
+{
+  assert_criteriont a;
+  const namespacet ns(goto_model.symbol_table);
```

Comment:
What about adding a member function to goto_model that returns a const ref to a namespace? That would allow to remove 143 lines (such as this one) from the code base.